dot-net-sdk/EppoClient.cs

```diff
@@ -70,75 +82,45 @@ public class EppoClient
             return null;
         }
 
-        var subjectVariationOverride = this.GetSubjectVariationOverride(subjectKey, configuration);
-        if (!subjectVariationOverride.IsNull())
-        {
-            return subjectVariationOverride;
-        }
-
         if (!configuration.enabled)
         {
             Logger.Info(
                 $"[Eppo SDK] No assigned variation because the experiment or feature flag {flagKey} is disabled");
             return null;
         }
 
-        var rule = RuleValidator.FindMatchingRule(subjectAttributes, configuration.rules);
-        if (rule == null)
+        var result = RuleValidator.EvaluateFlag(configuration, subjectKey, subjectAttributes);
+        if (result == null)
         {
             Logger.Info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
             return null;
         }
 
-        var allocation = configuration.GetAllocation(rule.allocationKey);
-        if (!IsInExperimentSample(subjectKey, flagKey, configuration.subjectShards, allocation!.percentExposure))
+        var assignment = result.Variation;
+
+        if (HasEppoValue.IsNullValue(assignment))
         {
-            Logger.Info("[Eppo SDK] No assigned variation. The subject is not part of the sample population");
+            Logger.Warn("[Eppo SDK] Assigned varition is null");
             return null;
         }
 
-        var assignedVariation =
-            GetAssignedVariation(subjectKey, flagKey, configuration.subjectShards, allocation.variations);
-        if (assignedVariation != null && !assignedVariation.IsNull())
+        try
         {
-            try
-            {
-                _eppoClientConfig.AssignmentLogger
-                    .LogAssignment(new AssignmentLogData(
-                        flagKey,
-                        rule.allocationKey,
-                        assignedVariation.StringValue() ?? "null",
-                        subjectKey,
-                        subjectAttributes
-                    ));
-            }
-            catch (Exception)
-            {
-                // Ignore Exception
-            }
+            _eppoClientConfig.AssignmentLogger
+                .LogAssignment(new AssignmentLogData(
+                    flagKey,
+                    result.AllocationKey,
+                    assignment.StringValue() ?? "null",
+                    subjectKey,
+                    subjectAttributes
```

Comment:
Seems like we're missing two fields here:
- the extra_logging fields coming from the UFC
- the metadata field

Python example: https://github.com/Eppo-exp/python-sdk/blob/main/eppo_client/client.py#L206-L215

Reply:
done.